metrics/web/lib/debug_menu/presentation/state/debug_menu_notifier.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:metrics/debug_menu/presentation/view_models/fps_monitor_local_config_view_model.dart';
+
+/// The [ChangeNotifier] that holds [LocalConfig]'s data and manages
+/// debug menu's features.
+class DebugMenuNotifier extends ChangeNotifier {
+  /// A [UseCase] that provides an ability to open the [LocalConfig] storage.
+  final OpenLocalConfigStorageUseCase _openLocalConfigStorageUseCase;
+
+  /// A [UseCase] that provides an ability to read the [LocalConfig].
+  final ReadLocalConfigUseCase _readLocalConfigUseCase;
+
+  /// A [UseCase] that provides an ability to update the [LocalConfig].
+  final UpdateLocalConfigUseCase _updateLocalConfigUseCase;
+
+  /// A [UseCase] that provides an ability to close the [LocalConfig] storage.
+  final CloseLocalConfigStorageUseCase _closeLocalConfigStorageUseCase;
+
+  /// Indicates whether the [LocalConfig] is loading.
+  bool _isLoading = false;
+
+  /// A view model that holds the [LocalConfig] data
+  /// for the FPS monitor feature.
+  FpsMonitorLocalConfigViewModel _fpsMonitorLocalConfigViewModel;
+
+  /// A [LocalConfig] containing the current configuration values.
+  LocalConfig _localConfig;
+
+  /// Holds the [PersistentStoreErrorMessage] that occured
+  /// during updating the [LocalConfig].
+  PersistentStoreErrorMessage _localConfigUpdatingError;
+
+  /// Returns `true` if the [LocalConfig] is loading.
+  /// Otherwise, returns `false`.
+  bool get isLoading => _isLoading;
+
+  /// Returns `true` if current [LocalConfig] is not `null`.
+  /// Otherwise, returns `false`.
+  bool get isInitialized => _localConfig != null;
+
+  /// A view model that provides the [LocalConfig] data
+  /// for the FPS monitor feature.
+  FpsMonitorLocalConfigViewModel get fpsMonitorLocalConfigViewModel =>
+      _fpsMonitorLocalConfigViewModel;
+
+  /// Provides an error description that occurred
+  /// during updating the [LocalConfig].
+  String get localConfigUpdatingErrorMessage =>
+      _localConfigUpdatingError?.message;
+
+  /// Creates a new instance of the [DebugMenuNotifier]
+  /// with the given parameters.
+  ///
+  /// Throws an [AssertionError] if any of the given parameters is `null`.
+  DebugMenuNotifier(
+    this._openLocalConfigStorageUseCase,
+    this._readLocalConfigUseCase,
+    this._updateLocalConfigUseCase,
+    this._closeLocalConfigStorageUseCase,
+  )   : assert(_openLocalConfigStorageUseCase != null),
+        assert(_readLocalConfigUseCase != null),
+        assert(_updateLocalConfigUseCase != null),
+        assert(_closeLocalConfigStorageUseCase != null);
+
+  /// Initializes the [LocalConfig].
+  Future<void> initializeLocalConfig() async {
+    try {
+      _setIsLoading(true);
+
+      await _openLocalConfigStorageUseCase();
+
+      final config = _readLocalConfigUseCase();
+
+      _setLocalConfig(config);
+
+      _setIsLoading(false);
+    } catch (_) {
+      initializeDefaults();
+    }
```

Comment:
```suggestion
    } catch (_) {
      initializeDefaults();
    } finally {
      _setIsLoading(false);
    }
```
And remove `_setIsLoading(false);` from the `initializeDefaults` method as it seems to be redundant there.